query-result.hh

```diff
@@ -224,20 +229,20 @@ public:
     }
 };
 
-inline future<result_memory_accounter> result_memory_limiter::new_mutation_read(query::max_result_size max_size, short_read short_read_allowed) {
-    return _memory_limiter.wait(minimum_result_size).then([this, max_size, short_read_allowed] {
-        return result_memory_accounter(result_memory_accounter::mutation_query_tag(), *this, max_size, short_read_allowed);
+inline future<result_memory_accounter> result_memory_limiter::new_mutation_read(query::max_result_size max_size, short_read short_read_allowed, const query_id& id) {
+    return _memory_limiter.wait(minimum_result_size).then([this, max_size, short_read_allowed, id] {
```

Comment:
since we get `id` by reference, we should capture it by reference,
otherwise we can just pass the query_id's around by value (which isn't that bad either).

Reply:
the `read_command` instance is not kept around, yes, that's why I store the id by value